### ISSUE TITLE
Best practice for Wasm NetConn 

### ISSUE THREAD

**Issue by fbaube** [nhooyr/websocket#423]
This is a support/documentation issue. 

When using the websocket library in a wasm client (in Safari), the stdlib `net.Conn` returned by `websocket.NetConn(..)` seems flakey.

Github issues for websocket mention more than once that for now and in the future (HTTP/2), the `net.Conn` is actually just an `io.ReadWriteCloser`. 

My wasm client has been using calls to `websocket/wsjson/Read(ctx,websocket.Conn,v)` and `websocket/wsjson/Write(ctx,websocket.Conn,v)`. 

Should my client instead be using the entities returned by `websocket.Conn.Reader(ctx)` and `websocket.Conn.Writer(ctx)` ?

**Comment by nhooyr** [comment]
I'm not sure what you're asking. Can you rephrase?

**Comment by fbaube** [comment]
In wasm in the browser (Safari), I want a protocol layer on which I can run the message-oriented protocol `(net/textproto)`. I think I want to use the API call `websocket.NetConn(..)`, and use the `net.Conn`that it returns. But this websocket `net.Conn` may or may not be a suitable target for development. 

In principle, if I have an `io.ReadWriteCloser`then I can call `net.textproto.NewConn(conn io.ReadWriteCloser) *Conn`, where the new `textproto.Conn`aggregates `textproto.Reader, textproto.Writer, textproto.Pipeline`.

Discussion: There's many issues and pull requests about this `net.Conn` that `websocket.NetConn(..)`returns -  the suitability of the `net.Conn`API for websockets, the desired-versus-actual functioning of Contexts and timeouts when calling into its API - so I wonder if my use case is realistic. I do not know.

So I guess my question is: Considering that when running in wasm, a `net.Conn` (as returned by `websocket.NetConn(..)`) is basically an `io.ReadWriteCloser`, which API might be the best target for `net/textproto` ? My goals here are not performance, but rather simplicity and robustness. 

- Should I use `websocket.NetConn(..)`and the resulting `net.Conn.Read(..)&Write(..)`? 
- Or should I work with the `io.Reader`and `io.WriteCloser`returned by `websocket.Reader(..)&Writer(..)`? 
- Or something else entirely ? 
- Or am I on the wrong track ? 

Many thanx

**Comment by nhooyr** [comment]
Use `websocket.NetConn` it will be fine for your use case. You're overthinking the other issues without understanding them. They do not apply here.
